### ISSUE TITLE
Creating a nested subject hierarchy breaks cloning

### ISSUE THREAD

**Issue by muratmaga** [Slicer/Slicer#7337]
## Summary

When a data object (e.g., pointlist) is nested under another data object, cloning functionality get completely broken (whether nested or not). This happens on Slicer r32267 on intel silicon mac

## Steps to reproduce

in a fresh scene download MRHead. 
See that you can successfully clone it by right click in Data module and choosing clone. 
Create a pointlist with a single point and drag it under cloned MRhead node. 
Now try to clone any object in the data scene, and notice the pasted error. 

Reproduces consistently. 

```
Switch to module:  "Data"
GetItemName: Failed to find subject hierarchy item by ID 31


GetItemName: Failed to find subject hierarchy item by ID 31


virtual QStandardItem *qMRMLSubjectHierarchyModelPrivate::insertSubjectHierarchyItem(vtkIdType, int) : Item mismatch when inserting subject hierarchy item with ID  38
```

**Comment by cpinter** [comment]
I followed the described steps in Slicer 5.5.0-2023-10-10 (r32228) and I was able to clone each item in the scene after the parenting, and there were no errors at all.
Same thing with Slicer 5.5.0-2023-10-30 r32252.

I tried it on Windows, but I don't see why it should be different on Mac. However, if someone has Mac please try to reproduce.

**Comment by lassoan** [comment]
Often these strange errors are related to SlicerMorph customizations. In this case, maybe it is due to enabling undo/redo.

**Comment by pieper** [comment]
I couldn't reproduce on mac with 5.4 (no SlicerMorph customizations).

**Comment by muratmaga** [comment]
> Often these strange errors are related to SlicerMorph customizations. In this case, maybe it is due to enabling undo/redo.

I don't think I have the SlicerMorph installed on it, since I was testing the lights. but will double check.

**Comment by cpinter** [comment]
I also tested it with 5.5.0-2023-11-02 r32276, and there were no issues. Just wanted to confirm that this is not something very recent affecting all platforms.

**Comment by muratmaga** [comment]
Re-opening the issues, as it turns out not related to nestedness or any extension, but the object. In current stable or previews on mac, cloning does not work model or markup data objects. In a fresh install, create a single pointList and try to clone.

**Comment by pieper** [comment]
I can replicate this with 5.4 and with my local build on mac.  Trying to clone a markup in a fresh start of Slicer fails, but if I download MRHead sample data the clone operation succeeds.  Oddly, the same operation succeeds on 5.4 on linux without needing to have other data loaded.

**Comment by jamesobutler** [comment]
Using Slicer 5.4.0 and Slicer 5.5.0-2023-11-04 on Windows, starting Slicer, creating a point list with a single defined point and then cloning it is successful.

Based on the above comments it appears this issue is unique to macOS?

**Comment by muratmaga** [comment]
> Trying to clone a markup in a fresh start of Slicer fails, but if I download MRHead sample data the clone operation succeeds.

You mean you can clone a markup or a model node after you have a volume loaded in the scene? In my case, it doesn't matter whether there is a volume on the scene or not. I can clone volumes and segmentations, but not markups or models. 

yes, I think this is mac specific. i don't have the issue on the linux or windows.

**Comment by pieper** [comment]
> You mean you can clone a markup or a model node after you have a volume loaded in the scene?

Yes, for me starting Slicer fresh and adding a markup, the markup cannot be cloned.

After adding a volume or even another markup then the cloning works.

**Comment by muratmaga** [comment]
By the way this error is persisting in the new stable 5.6 on macos.
